payment-handler/app-can-make-payment.js

```diff
@@ -0,0 +1,122 @@
+self.addEventListener('canmakepayment', event => {
+  if (event.methodData.length !== 1) {
+    const msg = 'Expected exactly one method data.';
+    event.respondWith(Promise.reject(new Error(msg)));
+    return;
+  }
+
+  const [method] = event.methodData;
+  if (!method || method.supportedMethods.length !== 1) {
+    const msg = 'Expected exactly one supported method name';
+    event.respondWith(Promise.reject(new Error(msg)));
+    return;
+  }
+
+  if (method.data.defaultParameter !== 'defaultValue') {
+    const msg = `Unexpected value for "defaultParameter": ${
+      method.data.defaultParameter
+    }`;
+    event.respondWith(Promise.reject(new Error(msg)));
+    return;
+  }
+
+  if ('defaultUnsupportedParameter' in method.data) {
+    const msg = 'Unexpected "defaultUnsupportedParameter"';
+    event.respondWith(Promise.reject(new Error(msg)));
+    return;
+  }
+
+  for (const attribute of ['total', 'displayItems']) {
```

Comment:
I'm actually not sure we need this test... this should be covered by the idlHarness tests.

Reply:
Removed. Please take another look.